tests/test_cli.py

```diff
@@ -98,25 +99,67 @@ def names_tar_file(name_xml):
     filename.unlink()  # delete created file
 
 
-def test_process(app, names_tar_file):
-    service = current_service_registry.get("names")
-    vocab_factory = get_vocabulary_config("names")
-    config = vocab_factory.get_config(origin=names_tar_file.absolute())
-    _process_vocab(config)
-    Name.index.refresh()
+@pytest.fixture
+def mock_config(names_tar_file):
+    return {
+        "readers": [
+            {
+                "type": "tar",
+                "args": {
+                    "origin": names_tar_file.absolute(),
+                    "regex": "\\.xml$",
+                },
+            },
+            {"type": "xml"},
+        ],
+        "transformers": [{"type": "orcid"}],
+        "writers": [
+            {
+                "type": "names-service",
+                "args": {
+                    "identity": system_identity,
+                },
+            }
+        ],
+    }
+
+
+def test_process(app, names_tar_file, mock_config):
+    with patch(
+        "invenio_vocabularies.factories.NamesVocabularyConfig"
+    ) as MockNamesVocabularyConfig:
```

Comment:
minor: instead of patching, shouldn't we test the new config directly in datastreams? In general, I see the CLI API as an end-user utility, while the reader/writer/transformer logic should be something that we can unit-test.

Reply:
Hmm I mocked it, because I want to avoid loading 1 day of ORCID data, streaming it from the bucket, which might be quite time consuming, so I simply keep the "original" readers, but we still test the ORCiD transformer and writer, the only thing we don't test is the ORCiD reader, let me know if you think we should change it anyway.

Reply:
Isn't this resolved now that we defaulted back to the original datastream config?